src/main/java/eu/dissco/core/digitalspecimenprocessor/service/MidsService.java

```diff
@@ -54,88 +57,175 @@ private boolean compliesToMidsTwo(DigitalSpecimen attributes) {
+      return null;
+    }
+  }
+
+  private boolean hasValue(String... terms) {
+    for (var term : terms) {
+      if (isValid(term)) {
+        return true;
+      }
+    }
+    return false;
   }
 
 
   private boolean compliesToMidsTwoPaleoBio(DigitalSpecimen attributes) {
-    return (attributes.getOdsMarkedAsType() != null && attributes.getOdsMarkedAsType())
+    return attributes.getOdsMarkedAsType() != null
         && isStratigraphyValid(attributes)
         && isQualitativeLocationValid(attributes)
         && isQuantitativeLocationValid(attributes);
   }
 
   private boolean isQuantitativeLocationValid(DigitalSpecimen digitalSpecimen) {
     if (locationIsPresent(digitalSpecimen)
-        && digitalSpecimen.getOccurrences().get(0).getLocation().getGeoreference() != null) {
-      var georeference = digitalSpecimen.getOccurrences().get(0).getLocation().getGeoreference();
-      return georeference.getDwcDecimalLatitude() != null
-          && georeference.getDwcDecimalLongitude() != null;
+        && digitalSpecimen.getDwcOccurrence().get(0).getDctermsLocation().getGeoReference()
+        != null) {
+      var location = digitalSpecimen.getDwcOccurrence().get(0).getDctermsLocation();
+      var georeference = location.getGeoReference();
+      var validValue = hasValue(location.getDwcLocationID(), georeference.getDwcFootprintWkt());
+      if (!validValue) {
+        validValue = georeference.getDwcDecimalLatitude() != null
+            && georeference.getDwcDecimalLongitude() != null;
+      }
+      return validValue;
     }
     return false;
   }
 
   private boolean occurrenceIsPresent(DigitalSpecimen attributes) {
-    return attributes.getOccurrences() != null && !attributes.getOccurrences().isEmpty()
-        && attributes.getOccurrences().get(0) != null;
+    return attributes.getDwcOccurrence() != null && !attributes.getDwcOccurrence().isEmpty()
+        && attributes.getDwcOccurrence().get(0) != null;
   }
 
   private boolean locationIsPresent(DigitalSpecimen digitalSpecimen) {
     return occurrenceIsPresent(digitalSpecimen)
-        && digitalSpecimen.getOccurrences().get(0).getLocation() != null;
+        && digitalSpecimen.getDwcOccurrence().get(0).getDctermsLocation() != null;
   }
 
   private boolean isQualitativeLocationValid(DigitalSpecimen digitalSpecimen) {
     if (locationIsPresent(digitalSpecimen)) {
-      var location = digitalSpecimen.getOccurrences().get(0).getLocation();
-      return isValid(location.getDwcContinent())
-          || isValid(location.getDwcCountry())
-          || isValid(location.getDwcCountryCode())
-          || isValid(location.getDwcCounty())
-          || isValid(location.getDwcIsland())
-          || isValid(location.getDwcIslandGroup())
-          || isValid(location.getDwcLocality())
-          || isValid(location.getDwcMunicipality())
-          || isValid(location.getDwcStateProvince())
-          || isValid(location.getDwcWaterBody());
+      var validValue = false;
```

Comment:
small thing but "validValue" sounds like it would contain a value, "isValidValue" sounds like it should be a boolean

Reply:
:+1: